### PR TITLE
Add simple pyproject parser and remove `requirements-detector` dependency

### PR DESCRIPTION
**Pull request by ljgray**

### CODE REVIEW

File: mkchimeenv.py [hunk 1]
```diff
@@ -106,6 +106,42 @@ def match_opcode(opcode: int) -> Tuple[int, str, bool]:
...
+    # Get the project file as a pathlib path
+    file = Path(path) / "pyproject.toml"
+
+    if not file.is_file():
+        raise FileNotFoundError(f"Project file {file} not found.")
+
+    data = toml.load(file)
+    project = data.get("project", {})
+    dependencies = project.get("dependencies", {}).copy()
+
```
Comment (ljgray) [inline mkchimeenv.py:133]: Question - should optional dependencies be included here?

File: mkchimeenv.py [hunk 2]
```diff
@@ -106,6 +106,42 @@ def match_opcode(opcode: int) -> Tuple[int, str, bool]:
         return (0, "Unknown", done)
 
 
+def find_requirements(path: str) -> List[Requirement]:
```
Comment (ljgray) [inline mkchimeenv.py:109]: Question - does this seem too simple?

File: mkchimeenv.py (hunk 1 above)
Comment (ketiltrout) [reply mkchimeenv.py:133]: No, at least not until we need to include some of them when running `mkchimeenv`.  Everything necessary should be getting installed without them.

File: mkchimeenv.py (hunk 2 above)
Comment (ketiltrout) [reply mkchimeenv.py:109]: I think it's fine.  It's a fairly simple thing we're trying to do now.  The hard part was all the build-system conversion.

**Review by ketiltrout — APPROVED** [review]

**Pull request merged**